### PR TITLE
chore: pnpm

### PR DESCRIPTION
**Pull request by ThangHuuVu**
## Reasoning 💡
Switch from yarn to [pnpm](https://pnpm.io/)
In this PR:
- Dependencies correction: Some adapter packages had the wrong/missing dependencies.
- `adapter-test `-> `@next-auth/adapter-test`
- `tsconfig` -> `@next-auth/tsconfig`
- In `release.yml`: Replace `actions/cache` with `actions/setup-node`
- dev app: Switch from npm-run-all to concurrently. It seems there is a bug in `npm-run-all` when using with `pnpm`. I couldn't run the command in parallel.
- remove `--no-workspaces` flag in `publish.ts`
## Checklist 🧢

- ~[ ] Documentation~
- ~[ ] Tests~
- [x] Ready to be merged

## Affected issues 🎟
None

### CODE REVIEW

File: .pnpm-debug.log [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (balazsorban44) [inline .pnpm-debug.log:1]: This file shoudl be gitignored

File: package.json [hunk 2]
```diff
@@ -38,6 +39,7 @@
     "eslint-plugin-promise": "^6.0.0",
     "git-log-parser": "1.2.0",
     "husky": "^7.0.4",
+    "pnpm": "^6.32.8",
```
Comment (balazsorban44) [inline package.json:42]: Is this necessary? I think we can expect it to be available globally

Comment (ThangHuuVu) [reply package.json:42]: Yeah these can definitely be dropped, thanks for the catch

**Comment by ThangHuuVu** [comment]
@balazsorban44 I replaced cache-node with setup-node v3. setup-node support built-in caching since pnpm v6.10+: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

**Pull request merged**